pyshacl/constraints/core/shape_based_constraints.py

```diff
@@ -110,9 +110,15 @@ def _evaluate_property_shape(
         if potentially_recursive and prop_shape in potentially_recursive:
             warn(ShapeRecursionWarning(_evaluation_path))
             return _non_conformant, _reports
-        if not prop_shape or not prop_shape.is_property_shape:
+        if not prop_shape:
             raise ReportableRuntimeError(
-                "Shape pointed to by sh:property does not exist or is not a well-formed SHACL PropertyShape."
+                f"SHACL PropertyShape not found: The shape referenced by sh:property does not exist. "
+                f"Please check if the shape '{prop_shape}' is defined."
```

Comment:
Sometimes property shapes are defined using blank nodes. Are we able to provide some smarts by going up the evaluation path until we find a shape defined with an IRI? This should significantly improve ill-formed shape debugging on a large shapes graph.

Reply:
While I may not be the ideal person to lead this effort, I'm interested in assisting or learning from someone more familiar with the project. I'd be happy to contribute in a supporting role or gain experience under guidance.

Reply:
@edmondchuc I believe that can be added as part of a subsequent PR. To avoid holding up this one.
There are some examples in some Validation Report Messages that have message generation code where a property-shape is pretty-printed taking into account if it is a BNode, the same logic can be implemented here.